### ISSUE TITLE
Sync folder gets put in confluence as a separate page

### ISSUE THREAD

**Issue by jvsteiner** [markdown-confluence/markdown-confluence#61]
So I finally tried out actually using a sync folder and I got some unexpected behavior.  maybe it is as intended, but it certainly wasn't the least surprising result for me.

What I intended:

1. Add a "DRAFT" page in confluence space to be a parent of new pages I wanted to make.
2. Create a "confluence-drafts" folder in my obsidian vault under which I would put new pages I wanted to publish for the first time before moving them elsewhere in the space
3. Set up the "DRAFTS" page ID in the global plug-in settings under the "Confluence Parent Page ID" setting
4. Add pages to the "confluence-drafts" folder in obsidian, and set it as the "Folder to publish" under global plug-in settings.
5. Publish all pages, after which they are expected to appear as direct children to the "DRAFTS" page in confluence

What actually happened:

At stage 5, in confluence, under "DRAFTS" there was a new page, called "confluence-drafts", and the pages I intended to publish were under that new page: "confluence-drafts"

**Comment by andymac4182** [comment]
That is the expected behaviour. I have been testing with a bunch of folders so I missed this. I will look at the root folder logic to make sure it can behave as expected.

**Comment by jvsteiner** [comment]
Interesting.  One consequence of using a parent ID to position the published pages is that you cannot publish at the root of the confluence space.  The behavior above means, additionally, that you cannot even publish at the second level in the space - at least not initially (although after you move them, you can publish anywhere)

**Comment by andymac4182** [comment]
I haven't tried but you could probably use the `Overview` page id as the parent ID to publish under and that will publish at the "root" of the confluence space. If you have a look at any page that appears at the root you can see it has a parent of the Overview page. 

Hopefully the second level issue won't be too hard. Guessing its related to the way I attempt to find the highest common folder and then publish everything under that. It must not be updating that common folder to the parent page id you provided.

**Comment by andymac4182** [comment]
If you edit the Overview page you can get the Page ID and set it as a the root and pages will be published to the root of the Space. 

The plugin finds the root of all pages in Obsidian that are being published and gives that folder the `Confluence Parent Page ID` from settings. Is it possible you have a markdown file outside that `Folder to Publish` setting that has `connie-publish: true` set?

Here is an example I set up 
* `Folder to Publish` is set to `Root Not Top/Confluence Pages` (This setting effectively just turned `connie-publish: true` as the default for all files in that folder.)

<img width="324" alt="image" src="https://user-images.githubusercontent.com/1297073/231562731-b09b926d-1899-40d9-82bd-b5f469551f48.png">
<img width="289" alt="image" src="https://user-images.githubusercontent.com/1297073/231562781-12cbebc1-d39b-472f-9641-a9e8a54250e0.png">

**Comment by andymac4182** [comment]
Separate to this. If you would like my Discord handle is `andymac4182#9680` be great to chat. I have started docs at https://obsidian-confluence.com/ would love some feedback or contributions (They are in the docs folder).

**Comment by jvsteiner** [comment]
have sent an invite from domoarrigato

**Comment by andymac4182** [comment]
I believe this is due to the folder root being calculated based on the local files not what is left in Confluence under the main folder. Will have a think about the best way to handle this. I think I will need to recalculate the root node based on the pages that are left unmoved. Not 100% sure how best to handle this at the moment.

**Comment by jvsteiner** [comment]
At the moment, I do the following:

1. Set the space id (overview page id) at the parent page id in the plugin settings
2. Create a "DRAFTS" folder in my obsidian root.
3. Set up the "Folder to publish" setting as "DRAFTS"
4. Publish, and then move the pages where I actually want them.

Personally, I don't think that many people will use the directory structure in obsidian to control the directory structure in confluence.  Most teams that actually need confluence have too many "cooks in the kitchen" for that, and anyway, it's very unclear that the same organization for confluence would work in Obsidian.  Obsidian really doesn't encourage a hierarchical directory structure, while confluence does.

**Comment by andymac4182** [comment]
Would it help if I introduced a setting around publish everything under the parent page id or structure the files based on the directory structure? So in your case it would just find all files not moved and put them directly under the parent page id regardless of where in your obsidian vault they are?

**Comment by jvsteiner** [comment]
I don't think it makes much of a practical difference - It's just a matter of which parent id I use, and how it is set up.  The result would be the same.  I tend to like to do any reorganizing in confluence only, and the current situation facilitates that with the `connie-dont-change-parent-page` option.  I just have to remember to set it.  Maybe it would be the tiniest bit easier if the default was true, rather than false, but don't have a compelling argument that this would be the case for most other users.
